packages/kit/src/runtime/client/renderer.js

```diff
@@ -174,12 +174,41 @@ export class Renderer {
 			if (!ready || !this.router) return;
 			this.session_id += 1;
 
-			const info = this.router.parse(new URL(location.href));
-			if (info) this.update(info, [], true);
+			this.invalidate(location.href);
 		});
 		ready = true;
 	}
 
+	init_listeners() {
+		/** @param {Event} event */
+		const trigger_prefetch = (event) => {
+			const a = find_anchor(event);
+			if (a && a.href && a.hasAttribute('sveltekit:prefetch')) {
+				this.prefetch(get_href(a));
+			}
+		};
+
+		/** @type {NodeJS.Timeout} */
+		let mousemove_timeout;
+
+		/** @param {MouseEvent|TouchEvent} event */
+		const handle_mousemove = (event) => {
+			clearTimeout(mousemove_timeout);
+			mousemove_timeout = setTimeout(() => {
+				// event.composedPath(), which is used in find_anchor, will be empty if the event is read in a timeout
+				// add a layer of indirection to address that
+				event.target?.dispatchEvent(
+					new CustomEvent('sveltekit:trigger_prefetch', { bubbles: true })
+				);
+			}, 20);
+		};
+
+		addEventListener('touchstart', trigger_prefetch);
+		addEventListener('mousemove', handle_mousemove);
+		addEventListener('sveltekit:trigger_prefetch', trigger_prefetch);
+		addEventListener('hashchange', () => this._update_page_store(new URL(window.location.href)));
```

Comment:
This need to be in that if (`hash_navigating`) that gets inited as false, and mutated to true in click event listener if the navigation only changes hash.

Reply:
Actually I just checked and you have duplicate for every event listener, so this is probably WIP still, or?

Reply:
Ummm. I'm pretty sure there is not a duplicate for every event listener unless I'm missing something. There are two `hashchange` listeners though, which I hadn't noticed. Perhaps they should be combined so that the handlers execute in a consistent order

Reply:
Yeah, I saw those 2 and assumed that there are more duplicates, sorry for that, but if that's the case then I guess we can merge them, if code should be after that other one for consistency.